### PR TITLE
Make correlation ID counter atomic

### PR DESCRIPTION
**Pull request by ychernysheva**
Fixes #[10218](https://adc.luxoft.com/jira/browse/FORDTCN-10218)

This PR is **ready** for review.

### Risk
This PR makes **no** API changes.

### Summary
There is a situation when messages are simultaneously created for different requests. During message creation process `GetNextHMICorrelationID`/`GetNextMobileCorrelationID` methods are invoked, and inside this methods corresponding counters are incremented. But now these counters aren't synchronized properly, because several threads may increment these counters at the same time, so `GetNextHMICorrelationID`/`GetNextMobileCorrelationID` methods may return same values for different requests. To avoid these situations type of this counters need to be changed from `uint32_t` to `atomic_uint_32_t`.

### CLA
- [x] I have signed [the CLA](https://docs.google.com/forms/d/e/1FAIpQLSdsgJY33VByaX482zHzi-xUm49JNnmuJOyAM6uegPQ2LXYVfA/viewform)

### CODE REVIEW

File: src/components/application_manager/src/message_helper/message_helper.cc [hunk 1]
```diff
@@ -3046,6 +3046,7 @@ void MessageHelper::SendOnStatusUpdate(const std::string& status,
 }
 
 void MessageHelper::SendGetSystemInfoRequest(ApplicationManager& app_mngr) {
+  SDL_LOG_AUTO_TRACE();
```
Comment (AKalinich-Luxoft) [inline src/components/application_manager/src/message_helper/message_helper.cc:3049]: @ychernysheva I think this is not related to fix and can be removed

Comment (ychernysheva) [reply src/components/application_manager/src/message_helper/message_helper.cc]: Debug is removed in 781d890e25f60d9a87b1f4bc8a8d1cecd8f857b0

**Review by LitvinenkoIra — APPROVED** [review]

File: src/components/application_manager/src/application_manager_impl.cc [hunk 2]
```diff
@@ -1388,7 +1387,7 @@ uint32_t ApplicationManagerImpl::GetNextMobileCorrelationID() {
     mobile_corelation_id_ = 0;
   }
 
-  return mobile_corelation_id_;
+  return mobile_corelation_id_.load();
```
Comment (LitvinenkoIra) [inline src/components/application_manager/src/application_manager_impl.cc:1390]: @ychernysheva @AKalinich-Luxoft do we need this change there? Wouldn't it work as a regular access operation if we do not specify the memory order?

Comment (ychernysheva) [reply src/components/application_manager/src/application_manager_impl.cc:1390]: @LitvinenkoIra I think that we can skip this `load()` call

Comment (AKalinich-Luxoft) [reply src/components/application_manager/src/application_manager_impl.cc:1390]: @ychernysheva please use either `load()` + `store()` or regular `operator=` and copy ctor.
Both implementations should have the same behavior.

Comment (ychernysheva) [reply src/components/application_manager/src/application_manager_impl.cc]: @LitvinenkoIra @AKalinich-Luxoft I remove `load()` call for consistency in a30f13f

**Review by AKalinich-Luxoft — APPROVED** [review]

**Review by VladSemenyuk — APPROVED** [review]
